Update cross build for ruby-2.3 and disable bundler usage for extconf.rb

larskanis opened the pull request:
The faked ruby environment for cross build doesn't work with bundler and
failed with the following error:

  /usr/local/rvm/rubies/ruby-2.3.0/bin/ruby -I. ../../../../ext/iobuffer/extconf.rb
  Could not find json-1.8.3 in any of the sources
  Run `bundle install` to install missing gems.
  rake aborted!

repeatedly commented:
I'm not sure what the problem is.
ruby 2.3.0 is latest stable so it should be supported.
Does this happen with rbenv?
If this problem happens with only rvm, I can't merge this PR.

larskanis commented:
Sorry for the unclear description! The problem happens only when calling
"rake build:windows" . It has nothing to do with rvm or rbenv. The cross
build does not use any non-stdlib libraries, so rubygems and bundler can be
disabled for this build process.

larskanis commented:
@repeatedly I now fixed the bundler issue while cross compilation per "rake build:windows" in rake-compiler-dock-0.5.1 and updated this PR accordingly.

repeatedly commented:
Awesome!
Thanks for updating the patch :+1:
